text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,484 @@
+})
+```
+
+CloudFront distributions use a default certificate (`*.cloudfront.net`) to support HTTPS by default. If you want to support HTTPS with your own domain
+name, you must associate a certificate with your distribution that contains your domain name. The certificate must be present in the AWS Certificate
+Manager (ACM) service in the US East (N. Virginia) region; the certificate may either be created by ACM, or created elsewhere and imported into ACM.
+
+```ts
+const myCertificate = new certmgr.DnsValidatedCertificate(this, 'mySiteCert', {
+  domainName: 'www.example.com',
+  hostedZone,
+});
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromBucket(myBucket),
+  certificate: myCertificate,
+});
+```
+
+Note that in the above example the aliases are inferred from the certificate and do not need to be explicitly provided.
+
+## Caching Behaviors
+
+Each distribution has a default cache behavior which applies to all requests to that distribution; additional cache behaviors may be specified for a
+given URL path pattern. Cache behaviors allowing routing with multiple origins, controlling which HTTP methods to support, whether to require users to
+use HTTPS, and what query strings or cookies to forward to your origin, among other behaviors.
+
+The properties of the default cache behavior can be adjusted as part of the distribution creation. The following example shows configuring the HTTP
+methods and viewer protocol policy of the cache.
+
+```ts
+const myWebDistribution = new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromHTTPServer({
+    domainName: 'www.example.com',
+    originProtocolPolicy: OriginProtocolPolicy.HTTPS_ONLY,
+  }),
+  behavior: {
+    allowedMethods: AllowedMethods.ALL,
+    viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  }
+});
+```
+
+Additional cache behaviors can be specified at creation, or added to the origin(s) after the initial creation. These additional cache behaviors enable
+customization for a specific set of resources based on a URL path pattern. For example, we can add a behavior to `myWebDistribution` to override the
+default time-to-live (TTL) for all of the images.
+
+```ts
+myWebDistribution.origin.addBehavior('/images/*.jpg', {
+  viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  defaultTtl: cdk.Duration.days(7),
+});
+```
+
+## Multiple Origins
+
+A distribution may have multiple origins in addition to the default origin; each additional origin must have (at least) one behavior to route requests
+to that origin. A common pattern might be to serve all static assets from an S3 bucket, but all dynamic content served from a web server. The
+following example shows how such a setup might be created:
+
+```ts
+const myWebsiteBucket = new s3.Bucket(...);
+const myMultiOriginDistribution = new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromWebsiteBucket(myBucket),
+  additionalOrigins: [
```

Comment:
> What is the order of evaluation of these expressions?

It's actually interesting you call that out here, because in this example it doesn't matter. There is a default origin with a default behavior created, and an additional origin with a behavior on '/api/*' created. You could create these in any order you wanted, and you'd end with the same result.

Reply:
Wait, does the default origin path map to `/` or `/*`?

Reply:
The default behavior has a path pattern of '*' (i.e., a catch-all). All paths not explicitly matching another behavior go to the default behavior. In the above case, no matter what order you define these objects in, all calls to /api/* will go to the additional origin, and all other calls will go to the default.